src/bijectors/truncated.jl

```diff
@@ -68,6 +68,27 @@ end
 
 with_logabsdet_jacobian(b::TruncatedBijector, x) = transform(b, x), logabsdetjac(b, x)
 
+# from stan https://github.com/stan-dev/math/blob/develop/stan/math/prim/constraint/lub_constrain.hpp etc
+function truncated_inv_logabsdetjac(x, a, b)
+    lowerbounded, upperbounded = isfinite(a), isfinite(b)
+    if lowerbounded && upperbounded
+        return log(b - a) - abs(x) + 2.0 * LogExpFunctions.log1pexp(-abs(x))
+    elseif lowerbounded
+        return x
+    elseif upperbounded
+        return x
```

Comment:
Generally, this branch can lead to type instabilities since in contrast to the first branch it does not depend on the types of `a` and `b` (e.g., consider `y::Float32`, `a::Int`, `b::Int`: the first branch returns a `Float64` whereas the other branches return a `Float32`; another example is `y::Int`, `a::Int`, `b::Int`).
There are multiple approaches to fix it, e.g., one could add a
```julia
   y, a, b = map(float, promote(y, a, b))
```
add the top of the function body.

These two branches can be simplified to
```suggestion
    elseif lowerbounded || upperbounded
        return y
```